components/BottomNavigation/examples/BottomNavigationExplicitlySetColorExample.swift

```diff
@@ -0,0 +1,145 @@
+    let color = UIColor(white: 0.2, alpha: 1)
+    appBar.headerViewController.headerView.backgroundColor = color
+    appBar.navigationBar.tintColor = .white
+    appBar.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.white]
+
+    commonBottomNavigationExampleInit()
+  }
+
+  @available(*, unavailable)
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+
+  func commonBottomNavigationExampleInit() {
+    view.backgroundColor = .lightGray
+    view.addSubview(bottomNavBar)
+
+    bottomNavBar.titleVisibility = .always
+
+    bottomNavBar.alignment = .centered
+
+    // Add items to the bottom navigation bar.
+    let tabBarItem1 = UITabBarItem(title: "Home", image: UIImage(named: "Home"), tag: 0)
+    let tabBarItem2 =
+      UITabBarItem(title: "Messages", image: UIImage(named: "Email"), tag: 0)
+    let tabBarItem3 =
+      UITabBarItem(title: "Favorites", image: UIImage(named: "Favorite"), tag: 0)
+    bottomNavBar.items = [ tabBarItem1, tabBarItem2, tabBarItem3 ]
+
+    // Select a bottom navigation bar item.
+    bottomNavBar.selectedItem = tabBarItem2;
+
+    // Layout buttons
+    redButton.frame = CGRect(x: view.center.x - 72, y: view.center.y - 72, width: 144, height: 56)
+    redButton.backgroundColor = .red
+    redButton.setTitle("Red theme", for: .normal)
+    redButton.addTarget(self, action: #selector(redTheme), for: .touchUpInside)
+    view.addSubview(redButton)
+
+    blueButton.frame = CGRect(x: view.center.x - 72, y: view.center.y + 16, width: 144, height: 56)
+    blueButton.backgroundColor = .blue
+    blueButton.setTitle("Blue theme", for: .normal)
+    blueButton.addTarget(self, action: #selector(blueTheme), for: .touchUpInside)
+    view.addSubview(blueButton)
+  }
+
+  func layoutBottomNavBar() {
+    let size = bottomNavBar.sizeThatFits(view.bounds.size)
+    let bottomNavBarFrame = CGRect(x: 0,
+                                   y: view.bounds.height - size.height,
+                                   width: size.width,
+                                   height: size.height)
+    bottomNavBar.frame = bottomNavBarFrame
+  }
+
+  override func viewWillLayoutSubviews() {
+    super.viewWillLayoutSubviews()
+    layoutBottomNavBar()
+  }
+
+  #if swift(>=3.2)
+  @available(iOS 11, *)
+  override func viewSafeAreaInsetsDidChange() {
+    super.viewSafeAreaInsetsDidChange()
+    layoutBottomNavBar()
+  }
+  #endif
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    appBar.addSubviewsToParent()
+
+    MDCBottomNavigationBarColorThemer.applySemanticColorScheme(colorScheme,
+                                                               toBottomNavigation: bottomNavBar)
+    bottomNavBar.backgroundColor = .darkGray
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    self.navigationController?.setNavigationBarHidden(true, animated: animated)
+  }
+
+  @objc func redTheme(_ button: UIButton) {
```

Comment:
In order to verify more code paths, please set a new set of items before the colors in this func and please set a new set of items after the colors in the bluefunc below.

Reply:
Setting up a different example for what we talked about

Reply:
Setting up a different example for when we explicitly reset items and then set their colors or when we set their colors then reset items.